### ISSUE TITLE
GAMClassifier hardcodes classes_=[0,1] and never encodes y: string/non-{0,1} labels silently mispredict (cross_val_score accuracy 0.0; sklearn contract violation)

### ISSUE THREAD

**Issue by SauersML** [SauersML/gam#389]
`GAMClassifier` (in `gamfit.sklearn`) hardcodes `self.classes_ = np.asarray([0, 1])` in `fit` and never inspects/encodes the supplied `y`. This breaks the scikit-learn classifier contract for any target whose labels are not literally the integers `{0, 1}` — most importantly **string labels** and `{-1,+1}` / `{1,2}` codings that sklearn users routinely pass.

The contract requires: (a) `classes_` reflects the **observed** labels, and (b) `predict` returns labels drawn from `classes_`. Neither holds. Worse, the failure is silent: a fit "succeeds" and predicts plausible-looking integers, but they are meaningless relative to the user's label space, so every downstream comparison (`score`, `cross_val_score`, `classification_report`) is wrong.

## Trigger
Construct `GAMClassifier`, fit on a binary target with non-`{0,1}` labels (e.g. strings), then `predict` / `cross_val_score`.

## Repro (deterministic, seed=0)
```python
import numpy as np, pandas as pd
from gamfit.sklearn import GAMClassifier
from sklearn.model_selection import cross_val_score

rng = np.random.default_rng(0); n = 200; x = rng.uniform(-2, 2, n)
y_int = (rng.uniform(size=n) < 1/(1+np.exp(-x))).astype(int)
y_str = np.where(y_int == 1, "pos", "neg")
X = pd.DataFrame({"x1": x})

clf = GAMClassifier(formula="y ~ s(x1)", family="binomial").fit(X, y_str)
print("classes_      :", clf.classes_)            # -> [0 1]
print("predict unique:", np.unique(clf.predict(X)))# -> [0 1]
print("cv accuracy   :", cross_val_score(clf, X, y_str, cv=3, scoring="accuracy"))  # -> [0. 0. 0.]
```

## Observed
```
classes_      : [0 1]
predict unique: [0 1]
cv accuracy   : [0. 0. 0.]
```
`cross_val_score` accuracy is `0.0` on every fold because `predict` emits ints `0/1` while the true labels are `"neg"/"pos"`, so every row mismatches. With `{-1,1}` or `{1,2}` integer labels the fit instead raises a misleading `InvalidConfigurationError` ("response is degenerate: all values are 1 — no non-events"), because the raw target is forwarded to a binomial family that treats any nonzero value as an event.

## Expected (sklearn contract)
- `classes_` should equal `np.unique(y)` (e.g. `array(['neg','pos'])`).
- `predict` should return labels from `classes_` (e.g. `'neg'`/`'pos'`).
- Internally the wrapper should label-encode `y` to `{0,1}` (positive class = `classes_[1]`) before fitting, and decode on `predict`.
- A standard sklearn classifier accepts any two-label target; this wrapper effectively only accepts a `{0,1}` int/float target.

## Root cause
`gamfit/sklearn.py`, `GAMClassifier.fit`:
```python
fitted = self._fit_model(X, y)
self.classes_ = np.asarray([0, 1])   # hardcoded; ignores actual y labels
return fitted
```
No `LabelEncoder`/`np.unique(y)` is applied, and `predict` does `self.classes_.take(argmax(...))`, so it can only ever return `0`/`1`. The fix is to derive `classes_` from the observed `y`, encode to the positive-class indicator before fitting, and validate that exactly two classes are present (or raise a clear error otherwise).

Confirmed present in current main: gamfit/sklearn.py:224 (HEAD 49366bfe7)

_Found via released-gamfit 0.1.135 fuzzing, confirmed against current main (wave 2)._

**Comment by SauersML** [comment]
🔒 claim: fix-all-issues workflow — agent for #389

**Comment by SauersML** [comment]
Fixed on main: commit 25392945d (`fix(sklearn) #389: encode arbitrary binary y instead of hardcoding classes_=[0,1]`) plus followup 299466e54.

GAMClassifier.fit now derives classes_ = np.unique(y), validates exactly two observed classes (clear ValueError otherwise), label-encodes the positive class (classes_[1]) to 1 before fitting the binomial GAM, and predict decodes back to the original label space via classes_.take(argmax(...)). String labels, {-1,+1}, {1,2}, and {0,1} all round-trip.

Regression tests on main in tests/test_python_api.py:
- test_sklearn_classifier_string_labels_roundtrip (the exact #389 repro: 'neg'/'pos', asserts classes_==['neg','pos'], predict returns those labels, in-sample accuracy > 0.7)
- test_sklearn_classifier_pm1_labels_roundtrip ({-1,+1} labels)

**Comment by SauersML** [comment]
🔓 release: fix-all-issues workflow — already fixed on main (25392945d + 299466e54): GAMClassifier derives classes_ from np.unique(y), encodes positive class to 1, decodes on predict; regression tests cover string and {-1,+1} labels.
